test/helper/resource.go

```diff
@@ -128,7 +128,21 @@ func NewStatefulSet(namespace string, name string) *appsv1.StatefulSet {
 }
 
 // NewService will build a service object.
-func NewService(namespace string, name string) *corev1.Service {
+// nodePort 0 means generate a random nodePort by k8s
+func NewService(namespace string, name string, svcType corev1.ServiceType, nodePort int32) *corev1.Service {
+	port := corev1.ServicePort{
+		Name:       "http",
+		Protocol:   corev1.ProtocolTCP,
+		Port:       80,
+		TargetPort: intstr.IntOrString{IntVal: 8080},
+	}
+	// specify a random nodePort explicitly rather than handing it over to K8S
+	// as create same a explicit nodePort service, a invalid error will occur
+	if svcType == corev1.ServiceTypeNodePort && nodePort != 0 {
```

Comment:
Hi @duanmengkk, IMOP, the `NewService` function is just a template and should not have too complex logic in it.
We can set a nodePort for NodePort type Service after we created a service.

Reply:
ok.i will fix it

Reply:
/cc @XiShanYongYe-Chang take a look,all reviews are fixed

Reply:
ok~